Python/errors.c

```diff
@@ -470,25 +470,43 @@ PyErr_Clear(void)
     _PyErr_Clear(tstate);
 }
 
+static PyObject*
+get_exc_type(PyObject *exc_value)  /* returns a borrowed ref */
+{
+    if (exc_value == NULL || exc_value == Py_None) {
+        return exc_value;
```

Comment:
Why not always return `Py_None`, like you do for the traceback? (At first I thought there was a bug here because you're returning the *value* from `get_exc_type()`. :-)